third_party/move/move-compiler-v2/src/plan_builder.rs

```diff
@@ -134,54 +119,67 @@ fn build_test_info<'func>(
                 let fn_msg = "Only functions defined as a test with #[test] can also have an \
                               #[expected_failure] attribute";
                 let abort_msg = "Attributed as #[expected_failure] here";
-                context.env.add_diag(diag!(
-                    Attributes::InvalidUsage,
-                    (fn_loc, fn_msg),
-                    (abort_attribute.loc, abort_msg),
-                ))
+                let abort_id = abort_attribute.node_id();
+                let abort_loc = env.get_node_loc(abort_id);
+                env.error_with_labels(&fn_id_loc, fn_msg, vec![(abort_loc, abort_msg.to_string())]);
             }
             return None;
         },
         Some(test_attribute) => test_attribute,
     };
 
+    let test_attribute_id = test_attribute.node_id();
+    let test_attribute_loc = env.get_node_loc(test_attribute_id);
+
+    let test_only_attribute_opt = attrs.iter().find(|a| a.name() == test_only_name);
+
     // A #[test] function cannot also be annotated #[test_only]
     if let Some(test_only_attribute) = test_only_attribute_opt {
+        const PREVIOUSLY_ANNOTATED_MSG: &str = "Previously annotated here";
```

Comment:
curious: why use `const` in some cases vs. `let` binding in others (eg, is there a difference between `PREVIOUSLY_..` and `msg` below that is worth differentiating in this pattern)?